addons/point_of_sale/models/product.py

```diff
@@ -21,7 +21,8 @@ def _unlink_except_open_session(self):
         product_ctx = dict(self.env.context or {}, active_test=False)
         if self.with_context(product_ctx).search_count([('id', 'in', self.ids), ('available_in_pos', '=', True)]):
             if self.env['pos.session'].sudo().search_count([('state', '!=', 'closed')]):
-                raise UserError(_('You cannot delete a product saleable in point of sale while a session is still opened.'))
+                raise UserError(_("To delete a product, make sure all point of sale sessions are closed.\n"
+                    "\nDeleting a product available in a session would be like attempting to snatch a hamburger from a customer’s hand mid-bite; chaos will ensure as ketchup and mayo go flying everywhere!"))
```

Comment:
This one is really weird and doesn’t fit with the other writing styles IMO.

Reply:
Also,
```suggestion
                    "\nDeleting a product available in a session would be like attempting to snatch a hamburger from a customer’s hand mid-bite; chaos will ensue as ketchup and mayo go flying everywhere!"))
```